scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -1370,6 +1401,526 @@ cdef class NumericalInversePolynomial(Method):
+        return (domain, order, u_resolution)
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    cdef inline void _ppf(self, const double *u, double *out, size_t N):
+        cdef:
+            size_t i
+        for i in range(N):
+            out[i] = unur_hinv_eval_approxinvcdf(self.rng, u[i])
+
+    def ppf(self, u):
+        """
+        ppf(u)
+
+        Approximated PPF of the given distribution.
+
+        Parameters
+        ----------
+        u : array_like
+            Qunatiles.
+
+        Returns
+        -------
+        ppf : array_like
+            Percentiles corresponding to given quantiles `u`.
+        """
+        u = np.asarray(u, dtype='d')
+        oshape = u.shape
+        u = u.ravel()
+        # UNU.RAN fills in ends of the support when u < 0 or u > 1 while
+        # SciPy fills in nans. Prefer SciPy behaviour.
+        cond0 = 0 <= u
+        cond1 = u <= 1
+        cond2 = cond0 & cond1
+        goodu = argsreduce(cond2, u)[0]
+        out = np.empty_like(u)
+        cdef double[::1] u_view = np.ascontiguousarray(goodu)
+        cdef double[::1] goodout = np.empty_like(u_view)
+        if cond2.any():
+            self._ppf(&u_view[0], &goodout[0], len(goodu))
+        np.place(out, cond2, goodout)
+        np.place(out, ~cond2, np.nan)
+        return np.asarray(out).reshape(oshape)[()]
+
+    def u_error(self, sample_size=100000):
+        """
+        u_error(sample_size=100000)
+
+        Estimate the u-error of the approximation using Monte Carlo simulation.
+        This is only available if the generator was initialized with a `dist`
+        object containing the implementation of the exact CDF under `cdf` method.
+
+        Parameters
+        ----------
+        sample_size : int, optional
+            Number of samples to use for the estimation. It must be greater than
+            or equal to 1000.
+
+        Returns
+        -------
+        max_error : float
+            Maximum u-error.
+        mean_absolute_error : float
+            Maximum absolute u-error.
+        """
+        # UNU.RAN doesn't return a proper error code for this condition.
+        if sample_size < 1000:
+            raise ValueError("`sample_size` must be greater than or equal to 1000.")
+        cdef double max_error, mae
+        cdef ccallback_t callback
+        _lock.acquire()
+        try:
+            self._messages.clear()
+            unur_set_stream(self._messages.handle)
+            init_unuran_callback(&callback, self._callback_wrapper)
+            self._check_errorcode(unur_hinv_estimate_error(self.rng, sample_size,
+                                                           &max_error, &mae))
+        finally:
+            _lock.release()
+            release_unuran_callback(&callback)
+        return UError(max_error, mae)
+
+    def qrvs(self, size=None, d=None, qmc_engine=None):
+        """
+        Quasi-random variates of the given RV.
```

Comment:
signature probably needs to be added to the docstring as for the other method

Reply:
done.